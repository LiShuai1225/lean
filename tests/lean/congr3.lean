universes l1 l2 l3
constants (A : Type.{l1}) (P : A → Type.{l2}) (P_ss : ∀ a, subsingleton (P a))
attribute P_ss [instance]
constants (Q : ∀ a, P a → Type.{l3}) (Q_ss : ∀ a (pa : P a), subsingleton (Q a pa))
attribute Q_ss [instance]
constant g (a : A) (pa : P a) : Q a pa → Prop
#congr_simp g
