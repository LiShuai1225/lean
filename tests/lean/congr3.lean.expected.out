[fixed, fixed, cast]
λ (a : A) (pa : P a) (a_1 : Q a pa), eq.refl (g a pa a_1)
:
∀ (a : A) (pa : P a) (a_1 : Q a pa), g a pa a_1 = g a pa a_1
